modules/ROOT/pages/clauses/load-csv.adoc

```diff
@@ -273,128 +604,327 @@ You can set the number of rows as in the example, where it is set to `500` rows.
+You may discover if this is the case by link:https://neo4j.com/docs/cypher-manual/current/query-tuning/#how-do-i-profile-a-query[profiling your queries], and attempt tweaking them to avoid it in case.
+
+
+=== Database heap and memory
+
+To help handle larger volumes of transactions, there are a few memory configuration settings you can tweak:
+
+* link:{neo4j-docs-base-uri}/operations-manual/{page-version}/configuration/configuration-settings/#config_server.memory.heap.initial_size[`server.memory.heap.initial_size`] and link:{neo4j-docs-base-uri}/operations-manual/{page-version}/configuration/configuration-settings/#config_server.memory.heap.max_size[`server.memory.heap.max_size`]: set to at least 4G.
+* link:{neo4j-docs-base-uri}/operations-manual/{page-version}/configuration/configuration-settings/#config_server.memory.pagecache.size[`server.memory.pagecache.size`]: ideally, a value large enough to keep the whole database in memory.
 
-Since `LOAD CSV` can temporary download a file to process it, it is important to note that `file()` will always return the path on disk.
-If `LOAD CSV` is invoked with a `file:///` URL that points to your disk `file()` will return that same path.
+// As a rule of thumb, you can create or update one million records in a single transaction per 2 GB of heap.
+////
+
+== Example
+
+====
+
+.Erase current database and import the full movie dataset
+[source, cypher]
+----
+// Clear data
+MATCH (n) DETACH DELETE n;
+
+// Create constraints
+CREATE CONSTRAINT Person_tmdbId IF NOT EXISTS
+FOR (p:Person)
+REQUIRE p.tmdbId IS UNIQUE;
+
+CREATE CONSTRAINT Movie_movieId IF NOT EXISTS
+FOR (m:Movie)
+REQUIRE m.movieId IS UNIQUE;
+
+// Create person nodes
+LOAD CSV WITH HEADERS FROM 'https://data.neo4j.com/importing-cypher/persons.csv' AS row
+MERGE (p:Person {tmdbId: toInteger(row.person_tmdbId)})
+SET p.name = row.name, p.born = date(row.born);
+
+// Create movie nodes
+LOAD CSV WITH HEADERS FROM 'https://data.neo4j.com/importing-cypher/movies.csv' AS row
+MERGE (m:Movie {id: toInteger(row.movieId)})
+SET m.title = row.title, m.imdbId = toInteger(row.movie_imdbId),
+    m.languages = split(row.languages, '|'), m.genres = split(row.genres, '|');
+
+// Create relationships
+LOAD CSV WITH HEADERS FROM 'https://data.neo4j.com/importing-cypher/acted_in.csv' AS row
+MATCH (p:Person {tmdbId: toInteger(row.person_tmdbId)})
+MATCH (m:Movie {id: toInteger(row.movieId)})
+MERGE (p)-[r:ACTED_IN]->(m)
+SET r.role = row.role;
+
+// Set additional node label
+MATCH (p:Person)-[:ACTED_IN]->()
+WITH DISTINCT p SET p:Actor;
+----
 
 .Result
-[role="queryresult"]
+[source, role="queryresult"]
 ----
-+------------------------------------------+
-| path                                     |
-+------------------------------------------+
-| "/home/example/neo4j/import/artists.csv" |
-+------------------------------------------+
-1 row
+Added 1 constraints
+Added 1 constraints
```

Comment:
```suggestion
Added 1 constraint.
Added 1 constraint.
```

Reply:
what about the following lines? (nodes, properties, labels)
i copied the output (well, parts of it) of cypher-shell - no periods at all, and plural for a single constraint. should i add a period to those lines as well?

Reply:
Ah from Cypher Shell, I was looking at Browser. Are all of the outputs from Cypher shell? If so, leave it, and I'll make a card for cypher shell to fix its English :P

Reply:
yes, i ran all the queries through cypher-shell to double-check :) alright, thanks